docs/copy-docs.py

```diff
@@ -0,0 +1,237 @@
+"""Copy docs from the napari repo into the napari.github.io repo
+
+python -m copy-docs.py [dstdir]
+"""
+
+import copy
+import os
+import os.path as osp
+import shutil
+import sys
+from fnmatch import fnmatch
+
+import yaml
+
+# path to copy and locations to copy to if different
+TO_COPY = [
+    'developers',
+    'community',
+    'release',
+    'roadmaps',
+    'images',
+    '_templates',
+    *[
+        (dire, osp.join(dire, 'stable'))
+        for dire in ('api', 'guides', 'plugins')
+    ],
+]
+
+# paths to ignore
+IGNORE = [
+    osp.join('images', 'logo.png'),
+]
+
+SRC = osp.dirname(__file__)
+
+DOC_EXTS = ['.md', '.rst', '.ipynb']
+
+TOC_IGNORE = ['release', 'api/stable', 'images', '_templates']
+
+
+def exclude_filter(path):
+    """Exclude files in the ignore list and duplicated files."""
+    for ignore in IGNORE:
+        if fnmatch(path, osp.join(SRC, ignore)):  # in ignore list
+            return True
+    else:
+        if osp.isdir(path) or osp.splitext(path)[1] != '.md':
+            return False
+        with open(path) as f:
+            firstline = f.readline()
+        return firstline.startswith('```{include}')  # duplicate file
```

Comment:
i'm curious, why does **```{include}`** indicate a duplicate file? 🤔

Reply:
basically to preserve the structure for the docs at https://napari.org/docs, I have files which are nothing more than
````markdown
```{include} filepath
```
````

that effectively point to where the files are actually located for the main https://napari.org site

so for example we used to have a file at `developer/roadmap_0_3` but now we want it at `roadmaps/0_3` so to preserve the old structure, `roadmaps/0_3` is not in the table of contents here, but it is in `napari.github.io`